### PR TITLE
Add Vimeo, Wistia and Image as new types

### PR DESCRIPTION
**Pull request by deracs**
Add 3 new media types - Image, Wistia & Vimeo

### CODE REVIEW

File: code/oEmbeddableFields.php [hunk 1]
```diff
@@ -7,387 +7,537 @@
...
+    {
 
-        $Media = array();
+        $Media = [];
+
+        if ($this->TypeEnabled('Image')) {
+            $Media['Image'] = UploadField::create('Image',
+                'Image');
+            $Media['Image']->setRightTitle(
+                'You must first save this Media before adding the image');
```
Comment (nathanbrauer) [inline code/oEmbeddableFields.php]: Condense each to one line.

File: code/oEmbeddableFields.php [hunk 2]
```diff
@@ -7,387 +7,537 @@
...
 	            </span>
-	            Slideshare URL',$this->owner->SlideshareID ?: null);
-	        $Media['SlideshareID']->setRightTitle(
-	            'Include the Slideshare <strong>URL</strong> for slideshows.<br/>'.
-	            '<strong>After saving:</strong> If valid, this will automatically convert to the Slideshare ID. If not, it will change to "0".'
-	        //TO GET IT: <u>Drag</u> this <a href="javascript:alert(typeof slideshare_object == \'object\' ? slideshare_object.presentationId : \'Not on a Slideshare page!\');">[SlideshareID]</a> to your bookmarks, then click the button while viewing a Slideshare page.'
-	        );
-        }else{
-	        $Media['SlideshareID'] = HiddenField::create('SlideshareID', '');
+	            Slideshare URL', $this->owner->SlideshareID ?: null);
```
Comment (nathanbrauer) [inline code/oEmbeddableFields.php]: If a call is too long to fit in one line "readably", each parameter should have its own line.

File: code/oEmbeddableFields.php [hunk 3]
```diff
@@ -7,387 +7,537 @@
...
 
-		if($this->TypeEnabled('YouTube')){
-	        $Media['YouTubeID'] = TextField::create('YouTubeID',
-	            '<span style="position:relative;">
+        if ($this->TypeEnabled('YouTube')) {
+            $Media['YouTubeID'] = TextField::create('YouTubeID',
+                '<span style="position:relative;">
 	                <a href="http://YouTube.com/upload" target="_blank" style="position:absolute;top:1.5em;white-space:nowrap;">Upload Video</a>
 	            </span>
 	            YouTube URL');
```
Comment (nathanbrauer) [inline code/oEmbeddableFields.php]: Each parameter should have its own line.

File: code/oEmbeddableFields.php [hunk 4]
```diff
@@ -7,387 +7,537 @@
...
+
+        return $json;
+
+    }
+
+    public function WistiaIdentifier($ID)
+    {
+
+        $link = 'http://home.wistia.com/medias/'.rawurlencode($ID);
+        ///https?:\/\/(.+)?(wistia.com|wi.st)\/(medias|embed)\/.*/
```
Comment (nathanbrauer) [inline code/oEmbeddableFields.php]: Do we need to keep this in?

**Review by nathanbrauer — CHANGES_REQUESTED** [review]
Please update README accordingly.

**Comment by deracs** [comment]
@nathanbrauer Updated those changes now

**Review by nathanbrauer — APPROVED** [review]

**Pull request merged**